code/modules/basketball/hoop.dm

```diff
@@ -0,0 +1,195 @@
+#define PICKUP_RESTRICTION_TIME 3 SECONDS // so other players can pickup the ball after someone scores
+
+/datum/crafting_recipe/basketball_hoop
+	name = "Basketball Hoop"
+	result = /obj/structure/hoop
+	reqs = list(/obj/item/stack/sheet/durathread = 5,
+				/obj/item/stack/sheet/iron = 1, // the backboard
+				/obj/item/stack/rods = 5)
+	time = 10 SECONDS
+	category = CAT_STRUCTURE
+
+/obj/structure/hoop
+	name = "basketball hoop"
+	desc = "Boom, shakalaka!"
+	icon = 'icons/obj/toys/basketball_hoop.dmi'
+	icon_state = "hoop"
+	anchored = TRUE
+	density = TRUE
+	layer = ABOVE_MOB_LAYER
+	/// Keeps track of the total points scored
+	var/total_score = 0
+	/// The chance to score a ball into the hoop based on distance
+	/// ex. a distance of two tiles away, throw_range_success[2], results in 80% chance to score
+	/// if someone shoots 3 tiles away (65% chance) or more, it scores 3 points
```

Comment:
For documents on single vars, the comment has to live on a single line or it'll get cut off on vsc's intellisense.

Reply:
Cutoff the excess comments with 0ff5b4781b8a27f350f462556ad4bb57c32d4645